mongorestore/restore.go

```diff
@@ -331,11 +332,29 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
-func stripInvalidOptions(indexes []IndexDocument) []IndexDocument {
+func convertLegacyIndexes(indexes []IndexDocument) []IndexDocument {
 	for i, index := range indexes {
+		for j, elem := range index.Key {
+			switch elem.Value {
+			case true:
+				md := bsonutil.MarshalD(indexes[i].Key)
+				indexJSON, err := md.MarshalJSON()
+				if err != nil {
+					indexJSON = []byte("<error formatting index>")
+				}
+				indexes[i].Key[j].Value = 1
+				log.Logvf(log.Always, "convertLegacyIndexes: converting the legacy vaule of '%s' from '%s' to '%s' in index '%s'", indexes[i].Key[j].Key, "true", "1", string(indexJSON))
```

Comment:
Rather than doing this logging element by element, I suggest just flagging if the Key changed and then logging the whole Key document at the end if so.